server/utils/licenceFactory.js

```diff
@@ -156,18 +159,23 @@ function injectVariablesIntoViewObject(fieldNames, fieldPositionObject, userInpu
     };
 }
 
-// Special case, doesn't follow normal rules
-function injectUserInputAppointmentAsObject(userInput, conditionText, userErrors) {
-    const {appointmentAddress, appointmentDate, appointmentTime} = userInput;
-    const {addressError, dateError, timeError, anyErrors} = getAppointmentErrors(userErrors);
+function injectMultiFieldsAsObject(userInput, conditionText, userErrors, config) {
+
+    const variableString = (variable, error) => error ? `[${error}]` : variable;
+
+    const strings = config.fields.map(fieldName => {
+        return variableString(getIn(userInput, [fieldName]), getIn(userErrors, [fieldName]));
+    });
 
-    const variableString = (error, variable) => error ? `[${error}]` : variable;
-    const addressString = variableString(addressError, appointmentAddress);
-    const dateString = variableString(dateError, appointmentDate);
-    const timeString = variableString(timeError, appointmentTime);
-    const string = `${addressString} on ${dateString} at ${timeString}`;
+    const fieldErrors = config.fields
+        .map(fieldName => getIn(userErrors, [fieldName]))
+        .filter(e => e);
```

Comment:
Absolutely fine, just by way of alternative you could avoid a second filter loop by using a reduce

Reply:
Maybe wouldn't be as readable though